### ISSUE TITLE
NVDA sometimes seems to miss entering spaces in typed text when "Announce formatting changes after the cursor" setting is enabled

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#1561]
**Reported by Palacee_hun on 2011-06-12 18:21**
I am a fast and quite accurate typist usually typing approx 150-300 chars/minute. Some months ago, when I began heavy text editings with NVDA, I noticed that spaces sometimes missed from my typed texts. First I attributed this to my eventual carelessness. But this kept on recurring and goes on to this very day. It even occurred nearly an hour ago when I composed my previous ticket. I have to double or triple check my important texts because of this. I can't say this is frequent, but I may not call it rare either.
I have done wast amounts of text composing and editing on various machines with other screen readers, most of them were much slower than my recent machine, among them were Pentium II class machines too. I have never ever had this sort of problem then, so after many many missing and hand-corrected spaces I began suspecting NVDA. That's why I report this. After all NVDA does a handful of special processing for the Space key, so there might lie the cause of this problem. I have excludedkeyboard defects as never has an other key ever been missed.

It's naturally hard to reproduce and investigate this, but you may try to type texts into edit boxes on webpages (like Gmail composing forms) or into MS Word 2003 Pro at the speeds close to mine, these are the ones I use most often, but this problem does not seem to be application-specific. I can't recall which was the first NVDA version with I noticed this problem.

**Comment by bhavyashah** [comment]
I am unable to replicate the reported issue with NVDA 2017.2 on a fairly powerful machine endowed with an I5 processor and 8 gb RAM. I tested this report by enabling the relevant setting, typing at 300-400 CPM in two or three breaks for a couple of minutes, and the text appeared properly spaced in all cases. Since we have no means of contacting Palacee_hun for additional diagnosis, I recommend closing this ticket as worksforme.

**Comment by ehollig** [comment]
Closing because of the reasoning in https://github.com/nvaccess/nvda/issues/1561#issuecomment-155290699 and the findings in https://github.com/nvaccess/nvda/issues/1561#issuecomment-320225395.
